src/components/tables/Table.tsx

```diff
@@ -0,0 +1,173 @@
+export type IAttributeNameOrValueFormatter = (attributeNameOrValue: string) => string;
+
+export interface ITableHeadingAttribute {
+  attributeName: string;
+  titleFormatter: IAttributeNameOrValueFormatter;
+  filterFormatter?: IAttributeNameOrValueFormatter; // use this for filter if you render JSX through the attribute formatter
+  sort?: boolean;
+  sortByMethod?: IAttributeNameOrValueFormatter;
+  attributeFormatter?: IAttributeFormatter;
+}
+
+export interface ITablePredicate {
+  props: IDropdownSearchProps;
+  attributeName: string;
+  attributeNameFormatter: IAttributeNameOrValueFormatter;
+};
+
+export interface ITableOwnProps extends React.ClassAttributes<Table> {
+  id: string;
+  headingAttributes: ITableHeadingAttribute[];
+  blankSlateDefault: IBlankSlateProps;
+  tableContainerClasses?: string[];
+  initialTableData?: ITableData;
+  collapsibleFormatter?: (tableRowData: ITableRowData, props: ITableProps) => JSXRenderable;
+  actionBar?: true | IActionBarProps;
+  getActions?: (rowData?: ITableRowData, props?: ITableProps) => IActionOptions[];
+  blankSlateNoResultsOnAction?: IBlankSlateProps;
+  blankSlateOnError?: IBlankSlateProps;
+  filter?: true | IFilterBoxProps;
+  filterMethod?: (attributeValue: any, props: ITableOwnProps) => boolean;
+  predicates?: ITablePredicate[];
+  navigation?: true | INavigationChildrenProps;
+  lastUpdatedLabel?: string;
+  manual?: (
+    tableOwnProps: ITableOwnProps,
+    shouldResetPage: boolean,
+    tableCompositeState: ITableCompositeState,
+    previousTableCompositeState: ITableCompositeState,
+  ) => IThunkAction;
+};
+
+export interface ItableCompositeStateProps {
+  readonly tableCompositeState?: ITableCompositeState;
+};
+
+export interface ITableProps extends ITableOwnProps, ItableCompositeStateProps, Partial<ITableDispatchProps> { }
+
+export class Table extends React.Component<ITableProps, any> {
+  private updateCountForLoadingBehavior: number = 0;
+  private isInitialLoad: boolean;
+
+  static defaultProps = {
+    tableCompositeState: { sortState: {} } as any,
+    initialTableData: DEFAULT_TABLE_DATA,
+  } as Partial<ITableOwnProps>;
+
+  constructor(props: ITableProps) {
+    super(props);
+
+    // Only use the initial load strategy for tables that do not provide initialTableData in their own props
+    this.isInitialLoad = props.initialTableData == DEFAULT_TABLE_DATA;
+  }
+
+  componentDidMount() {
+    if (this.props.onDidMount) {
+      this.props.onDidMount();
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.updateCountForLoadingBehavior < 2) {
+      this.updateCountForLoadingBehavior += 1;
+    } else {
+      /**
+       *  A count is utilized for a clean loading behavior during the initial load of the table
+       *  The first count occurs after mount
+       *  The second count occurs before the real data has loaded in the table
+       *  The initial load is than completed on the third update after the real data has loaded
+       */
```

Comment:
You could probably compare previous props and current props instead to figure out if the loading should turn off. The second time the component updates doesn't necessarily means it was triggered by the data being passed as props. Any state change or other prop could trigger an update.